web/views.py

```
@@ -6469,6 +6476,236 @@ def all_study_groups(request):
+
+    context = {
+        "plan": plan,
+        "billing_period": billing_period,
+        "stripe_public_key": settings.STRIPE_PUBLISHABLE_KEY,
+    }
+
+    return render(request, "checkout.html", context)
+
+
+@login_required
+def create_membership_subscription(request) -> JsonResponse:
```

Ruff (0.8.2)

`create_membership_subscription` is too complex (11 > 10)

(C901)